code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -520,76 +520,164 @@
+	name = "Morphine"
+	description = "A painkiller that allows the patient to move at full speed even in bulky objects. Extracted from poppies .Causes drowsiness and eventually unconsciousness in high doses. Overdose will cause a variety of effects, ranging from minor to lethal."
+	reagent_state = LIQUID
+	color = "#A9FBFB"
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+	overdose_threshold = 30
+	addiction_threshold = 25
+	opioid_power = 0.75
+
+/datum/reagent/medicine/opioid/morphine/on_mob_metabolize(mob/living/carbon/M)
+	..()
+	M.ignore_slowdown(type)
+
+/datum/reagent/medicine/opioid/morphine/on_mob_end_metabolize(mob/living/carbon/M)
+	M.unignore_slowdown(type)
+	..()
+
+/datum/reagent/medicine/opioid/morphine/overdose_process(mob/living/carbon/M)
 	if(prob(33))
 		M.drop_all_held_items()
-		M.adjustToxLoss(3*REM, 0)
-		. = 1
-		M.Dizzy(5)
-		M.Jitter(5)
+		M.Dizzy(2 * opioid_power)
+		M.Jitter(2 * opioid_power)
 	..()
 
+
+/datum/reagent/medicine/opioid/dimorphine
+	name = "Dimorphine"
+	description = "Opioid commonly known as heroin. Was created as a replacment for morphine, a lot more addictive but also able to be synthesized in labs. Overdose effects are harsher than morphine"
+	reagent_state = LIQUID
+	color = "#A9FBFB"
+	metabolization_rate = 0.75 * REAGENTS_METABOLISM
+	overdose_threshold = 15
+	addiction_threshold = 10
+	opioid_power = 1.5
+
+/datum/reagent/medicine/opioid/morphine/on_mob_metabolize(mob/living/carbon/M)
+	..()
+	M.ignore_slowdown(type)
+
+/datum/reagent/medicine/opioid/morphine/on_mob_end_metabolize(mob/living/carbon/M)
+	M.unignore_slowdown(type)
+	..()
+
+/datum/reagent/medicine/opioid/dimorphine/on_mob_life(mob/living/carbon/M)
+	M.set_drugginess(20)
+	M.Jitter(4)
+	..()
+
+/datum/reagent/medicine/opioid/dimorphine/overdose_process(mob/living/carbon/M)
+	if(prob(66))
+		M.drop_all_held_items()
+		M.Dizzy(2 * opioid_power)
+		M.Jitter(2 * opioid_power)
+		M.adjustOrganLoss(opioid_power,ORGAN_SLOT_HEART)
+	..()
+
+/datum/reagent/medicine/opioid/codeine
+	name = "Codeine"
+	description = "Weak opioid commonly used to treat minor discomforts, can be addicting. It's overdose effects are not as deadly as other drugs. It is very hard to overdose"
+	reagent_state = LIQUID
+	color = "#A9FBFB"
+	metabolization_rate = 0.75 * REAGENTS_METABOLISM
+	overdose_threshold = 75
+	addiction_threshold = 50
+	opioid_power = 0.1
+
+/datum/reagent/medicine/opioid/codeine/overdose_process(mob/living/carbon/M)
+	if(prob(16))
+		M.drop_all_held_items()
+		M.Dizzy( opioid_power)
+		M.Jitter(opioid_power)
+	..()
+
+/datum/reagent/medicine/opioid/fentanyl
+	name = "Fentanyl"
+	description = "An extremely powerful opioid , be careful when using it! When overdosed it causes toxin damage, "
+	reagent_state = LIQUID
+	color = "#64916E"
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+	overdose_threshold = 5
+	addiction_threshold = 4
+	opioid_power = 56.25 // fentanyl is 75 * morphine, so 75 * 7.5 = 56.25
```

Comment:
This number is fucking insane and should be dialed way the fuck back in the name of balance. Maybe 5, 6 at most. Can we see videos of it in game?

Reply:
fentanyl was ported from /datum/reagents/toxin, and is this crazy number because previously it did bad effects immidately , now there is an extremely small window of not - oding